src/app/schedule/schedule-settings/schedule-settings.component.ts

```diff
@@ -10,34 +10,33 @@ import { MomentService } from 'src/app/services/moment.service';
   styleUrls: ['./schedule-settings.component.css']
 })
 export class ScheduleSettingsComponent implements OnInit {
-  public settingsForm: FormGroup;
-  public monthsForSelect: Month[];
-
-  @Input() settings: Settings;
+  @Input() settings: Settings | undefined;
   @Output() settingsChange = new EventEmitter<Settings>();
 
+  public settingsForm: FormGroup | undefined;
+  public monthsForSelect: Month[] = [];
+
   constructor(private momentService: MomentService) {}
 
   ngOnInit() {
     this.initSettingsForm();
   }
 
   private initSettingsForm(): void {
-    let contractorNameInit: string;
-    if (this.settings && this.settings.contractorName) {
-      contractorNameInit = this.settings.contractorName;
-    } else {
-      // if contractor name not filled, check if user stored it before in a browser local storage
-      contractorNameInit = localStorage.getItem('contractorNameStored') || '';
-    }
+    const settingsContractorName =
+      this.settings && this.settings.contractorName;
```

Comment:
Nie wiem czy wiesz ale jako ciekawostka w TS 3.7 wprowadzili "optional chaining"
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

Próbowałem już migrować jeden projekt na 3.7 ale miałem problem z unit testami w "jest"(biblioteka do testowania).
Jak już angular wprowadzi pełne wsparcie to będziesz mógł tego typu przypisanie zapisać jako:
`this.settings?.contractorName;`

Jeśli się nudzisz to możesz spróbować zmienić na TS3.7 w package.json 😄

Reply:
Może wkrótce poeksperymentuję. Wiem, że się pojawił "optional chaining" w JS, nadal jako proposal, ale część przeglądarek już obsługuje. Dobrze wiedzieć, że w TS też już jest. 
Wiem, że taki zapis działa w angularze w template (chociaż nigdy nie pamiętałem czy właśnie w template czy w ts i zawsze musiałem sprawdzać na zasadzie "zadziała tu czy wywali błąd?" ;) ) W sumie to mega ułatwienie przy pozagnieżdżanych obiektach.